File: Arrays and Hashmap/contains-duplicate.cpp
#include <iostream>
#include <unordered_set>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> uset;
        for(int num : nums){
            if(uset.find(num) != uset.end())
                return true;
            uset.insert(num);
        }
        return false;
    }
};

int main() {
    Solution sol;

    vector<int> nums;
    nums.push_back(1);
    nums.push_back(2);
    nums.push_back(3);
    nums.push_back(4);
    nums.push_back(1);

    if(sol.containsDuplicate(nums)) {
        cout << "Duplicate found!" << endl;
    } else {
        cout << "No duplicates!" << endl;
    }

    return 0;
}

File: Arrays and Hashmap/contains-duplicate.py
'''
* Contains Duplicate: https://leetcode.com/problems/contains-duplicate/description/
'''

class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

# Example usage:
if __name__ == "__main__":
    solution = Solution()

    # Test Case 1
    nums1 = [1, 2, 3, 1]
    print("Input:", nums1)
    print("Output:", solution.containsDuplicate(nums1))  # True

    # Test Case 2
    nums2 = [1, 2, 3, 4]
    print("\nInput:", nums2)
    print("Output:", solution.containsDuplicate(nums2))  # False

    # Test Case 3
    nums3 = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2]
    print("\nInput:", nums3)
    print("Output:", solution.containsDuplicate(nums3))  # True
